pallets/moonbeam-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use frame_benchmarking::benchmarks;
+use pallet_xcm_benchmarks::{new_executor, XcmCallOf};
+use sp_std::vec;
+use sp_std::vec::Vec;
+use xcm::latest::prelude::*;
+
+benchmarks! {
+	// This benchmark does not use any additional orders or instructions. This should be managed
+	// by the `deep` and `shallow` implementation.
```

Comment:
I don't have any context for what "deep" and "shallow" mean here, although maybe I will after reviewing more...

Reply:
Me neither, its probably a copy paste from the polkadot comment. Will remove

Reply:
It seems this concept existed before, but now dissapeared https://github.com/paritytech/polkadot/blob/7a9f624777ad2d2adb3a1e6797a31f9d653c9587/xcm/xcm-builder/src/weight.rs#L30-L83.

I think it referred weighthing the instructions per se (shallow) or the instructions and its effects potentially in another chain (deep)

Reply:
Ah, thanks for digging that up :+1: